parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -489,7 +489,15 @@ configGPUDrivers() {
     retrycmd_if_failure_no_stats 120 5 25 update-initramfs -u || exit $ERR_GPU_DRIVERS_INSTALL_TIMEOUT
     wait_for_apt_locks
     {{/* if the unattened upgrade is turned on, and it may takes 10 min to finish the installation, and we use the 1 second just to try to get the lock more aggressively */}}
-    retrycmd_if_failure 600 1 3600 apt-get -o Dpkg::Options::="--force-confold" install -y nvidia-container-runtime="${NVIDIA_CONTAINER_RUNTIME_VERSION}+${NVIDIA_DOCKER_SUFFIX}" || exit $ERR_GPU_DRIVERS_INSTALL_TIMEOUT
+    retrycmd_if_failure 600 1 3600 apt-get -o Dpkg::Options::="--force-confold" install -y "cuda-drivers-fabricmanager-$GPU_FABRICMGR_VERSION" nvidia-container-runtime="${NVIDIA_CONTAINER_RUNTIME_VERSION}+${NVIDIA_DOCKER_SUFFIX}" || exit $ERR_GPU_DRIVERS_INSTALL_TIMEOUT
+    if [[ "${MIG_NODE}" == "true" ]]; then
+        retrycmd_if_failure 600 1 3600 apt-get -o Dpkg::Options::="--force-confold" install -y "cuda-drivers-fabricmanager-$GPU_FABRICMGR_VERSION" || exit $ERR_FABRIC_MANAGER_INSTALL_TIMEOUT
+    else
+        if dpkg -s "cuda-drivers-fabricmanager-$GPU_FABRICMGR_VERSION"; then
```

Comment:
Line 492 and 494 are almost the same except for the `nvidia-container-runtime`. So are we attempting to re-install `fabric-manager` when we are on a MIG node? That’s probably not the case though.

Reply:
Instead of removing fabric manager for non-MIG nodes, is it ok to only install it for MIG nodes? In that case, perhaps the installation time for non-MIG nodes would go down a bit too?

Reply:
oops-nice catch. L492 shouldn't have the fabric manager, only MIG nodes will have it.

since we don't have a mig specific vhd (just generic gpu), and there's minimal effect, I figured just tidying up the bits at runtime was ok. I think there needs to be a more general cleanup after this.

Reply:
got it, thanks!